awx_collection/plugins/modules/tower_inventory.py

```diff
@@ -119,6 +160,9 @@ def main():
         # If the state was absent we can let the module delete it if needed, the module will handle exiting from this
         module.delete_if_needed(inventory)
 
+    # Reset Org id to push in case copy_from was used.
+    org_id = module.resolve_name_to_id('organizations', organization)
```

Comment:
If you move the logic into the copy_item function this may not be needed.

Reply:
Moved, and your right